tests/fixtures/fixture.ts

```diff
@@ -95,3 +96,48 @@ export const onlyRoleToBeTagged = {
     value: "1118201414078976192",
   },
 };
+
+export const generateDummyRequestObject = ({
```

Comment:
The functions seems to be correct. However, there are a few potential errors or bugs that could occur.

The headers property is an object of key-value pairs. However, the code you provided is creating a new Map object for the headers property. This is unnecessary, as the headers property is already an object.
The code you provided is using the ?? operator to set the default values for the method, url, params, and query properties. However, the headers property does not have a default value. This means that if the headers property is not specified in the input object, the headers property will be undefined.
To fix these potential errors or bugs, you could do the following:

You could remove the new Map call from the headers property.
You could add a default value for the headers property, such as {}.

---Just for FYI---- no need to change anything here

Reply:
We need the map because if you see the actual handlers they use the `get` method to get the header values from the request object, this is only possible to do it in a map and not in an object. If I use a normal object then it will give errors.

Headers also have a default parameter `new Map(Object.entries(headers ?? {})),` just after the headers variable name.